hack/dockerized

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+set -e
+
+DIR="$(
+    cd "$(dirname "$0")"
+    pwd
+)"
+DOCKER_DIR=${DIR}/docker-builder
+KUBEVIRT_DIR="$(
+    cd "$DIR/../"
+    pwd
+)"
+
+BUILDER=${BUILDER_NAME:-kubevirtbuilder}
+RSYNCD_PORT=${RSYNCD_PORT:-10873}
+
+# Build the build container
+(cd ${DOCKER_DIR} && docker build . -q -t ${BUILDER})
+
+# Create the persistent docker volume
+if [ -z "$(docker volume list | grep ${BUILDER})" ]; then
+    docker volume create --name ${BUILDER}
+fi
+
+# Make sure that the output directory exists
+docker run -v "${BUILDER}:/root:rw,z" --rm -it ${BUILDER} mkdir -p /root/go/src/kubevirt.io/kubevirt/_out
```

Comment:
It's unclear to me why we need this persistent volume and why we're rsyncing to it. At first glance this all seems a bit more complicated than it needs to be. maybe I'm missing something?

If I was approaching building kubevirt assents using a docker container, I'd just bind mount the src tree directly into the docker container and then run the normal build steps in the container rather than directly on the host. The container would have all the build tools, but it would be working directly on the src tree in the host file system.

Reply:
> If I was approaching building kubevirt assents using a docker container, I'd just bind mount the src tree directly into the docker container and then run the normal build steps in the container rather than directly on the host. The container would have all the build tools, but it would be working directly on the src tree in the host file system.

I had exactly that before, but there are a few issues with it (I erased the commit history of it in this PR). Issues are:
 * It is not possible to bind-mount with the right uid/gid
 * It is then extremely complex to synchronize uids und gids
 * Bind mounts are extremely slow. Surprisingly rsyncing in- and out and building inside is much-much faster (except from the rsync overhead pretty much the same like direct building)
 * With the docker volume in place, a developer gets dependency caching for free
 * It works pretty nice, k8s does it too

Reply:
- If you run the container as your UID/GID it just works, there is no uid/gid sync you need to do. That's how I do it. Maybe your docker daemon is missing something?
- How are bind mounts slow? Aren't you missing some kernel functionality? Do you mean the mounting is slow or the access to thos files is then slow? What filesystem are you using?